harvester_e2e_tests/fixtures/api_client.py

```diff
@@ -220,6 +220,13 @@ def login(self, ipaddr, port=22, jumphost=False, **kwargs):
                 cli.set_missing_host_key_policy(MissingHostKeyPolicy())
                 kws = dict(username=self.username, password=self.password, pkey=self.pkey)
                 kws.update(kwargs)
+
+                # in case we're using a password to log into the host, this
+                # prevents paramiko from getting confused by ssh keys in the ssh
+                # agent:
+                if self.password and not self.pkey:
+                    kws.update(dict(allow_agent=False, look_for_keys=False))
```

Comment:
The implement will implicit overwrite `allow_agent` and `look_for_keys` when call the method as
```python
login("IP", allow_agent=True, look_for_keys=True)
```

So please update the method signature to add parameters `allow_agent` and `look_for_keys` with default `False` and merge them into `kws`.

Reply:
Ok done, but calling `login("IP", allow_agent=True, look_for_keys=True)` makes no sense because the authentication will still fail if they want to use password authentication but have an SSH key in their ssh agent.

Keep in mind that some people have an SSH key in their agent that has nothing to do with these tests.